Modularized version of PR 60936 (#61428)
openshift/openshift-docs: modules/oadp-ceph-cephfs-back-up.adoc

// Module included in the following assemblies:
//
// * backup_and_restore/application_backup_and_restore/backing_up_and_restoring/backing-up-applications.adoc

ifeval::["{context}" == "cephfs"]
:cephfs:
endif::[]
ifeval::["{context}" == "split"]
:split:
endif::[]

:_content-type: PROCEDURE
[id="oadp-ceph-cephfs-back-up_{context}"]

ifdef::cephfs[]
= Backing up data using OADP 1.2 Data Mover and CephFS storage
You can use OpenShift API for Data Protection (OADP) 1.2 Data Mover to back up data using CephFS storage by enabling the shallow copy feature of CephFS storage.

> endif::cephs[]

Reviewer: ```suggestion
endif::cephfs[]
```

> ifdef::cephfs[]
> = Backing up data using OADP 1.2 Data Mover and CephFS storage
> You can use OpenShift API for Data Protection (OADP) 1.2 Data Mover to back up data using CephFS storage by enabling the shallow copy feature of CephFS storage.
> endif::[]
> 
> ifdef::split[]
> = Backing up data using OADP 1.2 Data Mover and split volumes
> You can use OpenShift API for Data Protection (OADP) 1.2 Data Mover to back up data in an environment that has split volumes.
> endif::[]

Reviewer: idr the rules about this, but I see ones in the repo seem to have the condition spelled out in the endif for the body text like this
```suggestion
ifdef::cephfs[]
= Backing up data using OADP 1.2 Data Mover and CephFS storage
You can use OpenShift API for Data Protection (OADP) 1.2 Data Mover to back up data using CephFS storage by enabling the shallow copy feature of CephFS storage.
endif::[]

ifdef::split[]
= Backing up data using OADP 1.2 Data Mover and split volumes
You can use OpenShift API for Data Protection (OADP) 1.2 Data Mover to back up data in an environment that has split volumes.
endif::split[]
```

.Procedure

> . Create a `Backup` CR similar to the following:

Reviewer 2: ```suggestion
. Create a `Backup` CR similar to the following example:
```

+
.Example `Backup` CR
+
[source,yaml]
----
apiVersion: velero.io/v1
kind: Backup

> metadata:
>   name: <backup-name>
>   namespace: <protected-ns>
> spec:
>   includedNamespaces:
>   - <app-ns>

Reviewer 2: ```suggestion
metadata:
  name: <backup_name>
  namespace: <protected_ns>
spec:
  includedNamespaces:
  - <app_ns>
```

  storageLocation: velero-sample-1
----

> . Monitor the progress of the `VolumeSnapshotBackup` CRs by doing the following:

Reviewer 2: Just a nit from the [ISG guidance](https://www.ibm.com/docs/en/ibm-style?topic=format-procedures#introducing-the-procedure) on lead-in sentences: 
```suggestion
. Monitor the progress of the `VolumeSnapshotBackup` CRs by completing the following steps:
```

.. To check the progress of all the `VolumeSnapshotBackup` CRs, run the following command:
+
[source, terminal]
----
$ oc get vsb -n <app-ns>
----

.. To check the progress of a specific `VolumeSnapshotBackup` CR, run the following command:
+
[source,terminal]
----

> $ oc get vsb <vsb-name> -n <app-ns> -ojsonpath="{.status.phase}`

Reviewer 2: ```suggestion
$ oc get vsb <vsb_name> -n <app_ns> -ojsonpath="{.status.phase}`
```

----

. Wait several minutes until the `VolumeSnapshotBackup` CR has the status `Completed`.

> . Verify that there is at least one snapshot in the object store that is given in the Restic `secret`. You can check for this snapshot in your targeted `BackupStorageLocation` storage provider that has a prefix of `/<OADP-namespace>`.

Reviewer 2: ```suggestion
. Verify that there is at least one snapshot in the object store that is given in the Restic `secret`. You can check for this snapshot in your targeted `BackupStorageLocation` storage provider that has a prefix of `/<OADP_namespace>`.
```
